schema/shared/src/main/scala-3/zio/blocks/schema/DerivedOptics.scala

```diff
@@ -0,0 +1,199 @@
+package zio.blocks.schema
+
+import scala.quoted.*
+import scala.language.dynamics
+
+/**
+ * A trait that can be extended by companion objects to automatically derive
+ * optics (Lens for case class fields, Prism for sealed trait/enum variants).
+ *
+ * Usage:
+ * {{{
+ * case class Person(name: String, age: Int)
+ * object Person extends DerivedOptics[Person]
```

Comment:
I think you could show how `export` can be used to make the optics top-level in Scala 3.

e.g. 

```scala
object Person extends DerivedOptics[Person] {
  export optics.*
}
```